pandas/core/algorithms.py

```diff
@@ -1558,6 +1559,81 @@ def take_nd(arr, indexer, axis=0, out=None, fill_value=np.nan, mask_info=None,
 take_1d = take_nd
 
 
+def take_ea(arr, indexer, fill_value=_default_fill_value):
```

Comment:
I threw this in `core/algorithms.py`. Does it make more sense in `core/arrays/base.py`, either as a standalone method or as a staticmethod on ExtensionArray?

Reply:
this ended up exposed in pd.api.extensions but it isn't clear to me why.  are we sure its needed there?